docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+         * @Transform /^"([^"]+)" country$/
+         */
+        public function getCountryByName($countryName) // Thanks to this method we got in our ManagingCountries an Country object.
+        {
+            $countryCode = $this->countryNameConverter->convertToCode($countryName);
+            $country = $this->countryRepository->findOneBy(['code' => $countryCode]);
+
+            if (null === $country) {
+                throw new \InvalidArgumentException(sprintf('Country with name %s does not exist', $countryName));
+            }
+
+            return $country;
+        }
+    }
+
+Setup contexts
+~~~~~~~~~~~~~~
+For setup context we need different scenario with more background steps and all preparing scene steps.
+Editing scenario will be great for this example:
+
+.. code-block:: feature
+        Given the store has disabled country "France"
+        And I want to edit this country
+        When I enable it
+        And I save my changes
+        Then I should be notified that it has been successfully edited
+        And this country should be enabled
+
+.. code-block:: php
+    namespace Sylius\Behat\Context\Setup;
+
+    final class GeographicalContext implements Context
+    {
+        /**
+         * @var SharedStorageInterface
+         */
+        private $sharedStorage;
+
+        /**
+         * @var FactoryInterface
+         */
+        private $countryFactory;
+
+        /**
+         * @var RepositoryInterface
+         */
+        private $countryRepository;
+
+        /**
+         * @var CountryNameConverterInterface
+         */
+        private $countryNameConverter;
+
+        /**
+         * @param SharedStorageInterface $sharedStorage
+         * @param FactoryInterface $countryFactory
+         * @param RepositoryInterface $countryRepository
+         * @param CountryNameConverterInterface $countryNameConverter
+         */
+        public function __construct(
+            SharedStorageInterface $sharedStorage,
+            FactoryInterface $countryFactory,
+            RepositoryInterface $countryRepository,
+            CountryNameConverterInterface $countryNameConverter
+        ) {
+            $this->sharedStorage = $sharedStorage;
+            $this->countryFactory = $countryFactory;
+            $this->provinceFactory = $provinceFactory;
+            $this->countryRepository = $countryRepository;
+            $this->countryNameConverter = $countryNameConverter;
+            $this->countryManager = $countryManager;
+        }
+
+        /**
+         * @Given /^the store has disabled country "([^"]*)"$/
+         */
+        public function theStoreHasDisabledCountry($countryName) // This method save country in data base.
+        {
+            $country = $this->createCountryNamed(trim($countryName));
+            $country->disable();
+
+            $this->sharedStorage->set('country', $country);
+            // Shared storage is an helper service for transferring objects between steps.
```

Comment:
IMO this section is precise enough.
http://docs.sylius.org/en/latest/behat/how-to-use-transformers.html